File: 공부내용정리/알고리즘/programmers/HashMap/Clothes/Clothes.java

import java.util.Map;
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        
        Map<String, Integer> clothesMap = new HashMap<String, Integer>();
        int result = 1;
        
        for ( String[] clothes_data : clothes) {
            String type = clothes_data[1];
            if ( clothesMap.containsKey(type) ) {
                clothesMap.put(type, clothesMap.get(type) + 1);
            } else {
                clothesMap.put(type, 1);
            }
        }
                        
        for (int types : clothesMap.values() ) {
            result = result * ( types + 1 );
        }
        
        return result - 1;
        
    }
}

File: 공부내용정리/알고리즘/programmers/HashMap/Clothes/Clothes.js
function solution(clothes) {
  // 최초 1
  let answer = 1;
  const clothesMap = new Map();
  
  clothes.forEach((data) => {
      const type = data[1];
      if (!clothesMap.has(type)) {
          clothesMap.set(type, 1);   
      } else {
          clothesMap.set(type, clothesMap.get(type) + 1)
      }   
  })
  
  for( const methods of clothesMap.values()) {
      answer *= methods + 1;
  }

  
  // 모두 안입었을 때
  return answer - 1;
}